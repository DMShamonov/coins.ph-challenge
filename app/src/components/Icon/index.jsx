// @flow

import * as React from 'react';
import classnames from 'classnames';

import type { PropsType } from './types.js.flow';

export default function Icon({
  theme,
  width,
  height,
  title,
  className,
}: PropsType): React.Node {
  return (
    <span
      className={classnames('icon', {
        [`icon_theme_${(theme: any)}`]: theme,
      }, className)}
      style={{
        width: width || height,
        height: height || width,
        // $FlowFixMe
        backgroundSize: (width || height) && `${width || height}px ${height || width}px`,
      }}
      title={title}
    />
  );
}
