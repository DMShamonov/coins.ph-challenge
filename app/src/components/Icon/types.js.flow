// @flow

export type IconThemeType = (
  | 'avatar-placeholder'
  | 'btc'
  | 'btc-wallet'
  | 'close'
  | 'logout'
);

export type PropsType = {|
  theme: IconThemeType,
  width?: number,
  height?: number,
  title?: string,
  className?: string
|}
