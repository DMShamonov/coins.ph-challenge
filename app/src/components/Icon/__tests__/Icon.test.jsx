import React from 'react';
import { shallow } from 'enzyme';

import Icon from '../index';

describe('`Icon` component should render correctly', () => {
  test('with default props', () => {
    expect(shallow(<Icon theme="avatar-placeholder" />)).toMatchSnapshot();
  });

  test('with `width` prop', () => {
    expect(shallow(<Icon theme="avatar-placeholder" width={48} />)).toMatchSnapshot();
  });

  test('with `height` prop', () => {
    expect(shallow(<Icon theme="avatar-placeholder" height={48} />)).toMatchSnapshot();
  });

  test('with both `width` and `height` props', () => {
    expect(shallow(<Icon theme="avatar-placeholder" width={48} height={48} />)).toMatchSnapshot();
  });

  test('with `className` prop', () => {
    expect(shallow(<Icon theme="avatar-placeholder" className="__CLASSNAME__" />)).toMatchSnapshot();
  });
});
