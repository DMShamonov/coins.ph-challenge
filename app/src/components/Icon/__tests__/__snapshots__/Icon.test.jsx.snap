// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`Icon\` component should render correctly with \`className\` prop 1`] = `
<span
  className="icon icon_theme_avatar-placeholder __CLASSNAME__"
  style={
    Object {
      "backgroundSize": undefined,
      "height": undefined,
      "width": undefined,
    }
  }
/>
`;

exports[`\`Icon\` component should render correctly with \`height\` prop 1`] = `
<span
  className="icon icon_theme_avatar-placeholder"
  style={
    Object {
      "backgroundSize": "48px 48px",
      "height": 48,
      "width": 48,
    }
  }
/>
`;

exports[`\`Icon\` component should render correctly with \`width\` prop 1`] = `
<span
  className="icon icon_theme_avatar-placeholder"
  style={
    Object {
      "backgroundSize": "48px 48px",
      "height": 48,
      "width": 48,
    }
  }
/>
`;

exports[`\`Icon\` component should render correctly with both \`width\` and \`height\` props 1`] = `
<span
  className="icon icon_theme_avatar-placeholder"
  style={
    Object {
      "backgroundSize": "48px 48px",
      "height": 48,
      "width": 48,
    }
  }
/>
`;

exports[`\`Icon\` component should render correctly with default props 1`] = `
<span
  className="icon icon_theme_avatar-placeholder"
  style={
    Object {
      "backgroundSize": undefined,
      "height": undefined,
      "width": undefined,
    }
  }
/>
`;
